src/messages/Selection.hpp

```diff
@@ -72,6 +73,13 @@ struct Selection {
         return !this->operator==(b);
     }
 
+    //union of both selections
+    Selection operator|=(const Selection &b) const
```

Comment:
This should probably be `operator|`, since it doesn't assign.

```suggestion
    Selection operator|(const Selection &b) const
```

Note that this messes with `start` and `end` a bit, which denote the start/end point of the selection. When double-clicking, however, we have two starting positions (word start & end), so these become meaningless, but there's nothing using them for double-click-selection right now.

Reply:
op type changed 008fa5b5ad36782c835b2451f43d9b22cc602099, 
can you elaborate on issue with start/end? would be best to have it work in all scenarios, even if currently unused